src/helpers/attributes.ts

```diff
@@ -47,3 +48,18 @@ export const callbackConverter = (value: string | ((...params: any) => any)) =>
 
   return value;
 };
+
+export const enumConverter = <T extends Enum>(enumValues: T) => {
+  return (value: string): T[keyof T] | undefined => {
+    // we compare the requested key, and the enums keys as lowercase so that
+    // the attribute that is exposed to the user is case insensitive
+    const lowercaseKey = value.toLowerCase();
+    const enumKey = Object.keys(enumValues).find(
+      (key) => key.toLowerCase() === lowercaseKey,
```

Comment:
Not that it matters in this case, but I'd generally recommend `localeCompare`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#comparing_strings